bindings/python/YarpUtilities/src/VectorsCollection.cpp

```diff
@@ -46,6 +47,69 @@ void CreateVectorsCollectionServer(pybind11::module& module)
              })
         .def("prepare_data", &VectorsCollectionServer::prepareData);
 }
+
+void CreateVectorsCollectionClient(pybind11::module& module)
+{
+    namespace py = ::pybind11;
+
+    using namespace ::BipedalLocomotion::YarpUtilities;
+
+    py::class_<VectorsCollectionClient>(module, "VectorsCollectionClient")
+        .def(py::init())
+        .def(
+            "initialize",
+            [](VectorsCollectionClient& impl,
+               std::shared_ptr<const ::BipedalLocomotion::ParametersHandler::IParametersHandler>
+                   handler) -> bool { return impl.initialize(handler); },
+            py::arg("handler"))
+        .def("connect", &VectorsCollectionClient::connect)
+        .def("disconnect", &VectorsCollectionClient::disconnect)
+        .def("get_metadata",
+        [](VectorsCollectionClient& impl) -> BipedalLocomotion::YarpUtilities::VectorsCollectionMetadata
+        {
+            BipedalLocomotion::YarpUtilities::VectorsCollectionMetadata metadata;
+            impl.getMetadata(metadata);
+            return metadata;
+        })
+        .def("read_data",
+             [](VectorsCollectionClient& impl, bool shouldWait) -> std::map<std::string, std::vector<double>>
+             {
+                BipedalLocomotion::YarpUtilities::VectorsCollection* collection = impl.readData(shouldWait);
+                return collection->vectors;
+             });
+}
+
+void CreateVectorsCollectionMetadata(pybind11::module& module)
+{
+    namespace py = ::pybind11;
+
+    using namespace ::BipedalLocomotion::YarpUtilities;
+
+    py::class_<VectorsCollectionMetadata>(module, "VectorsCollectionMetadata")
+        .def(py::init())
+        .def(py::init<const std::map<std::string, std::vector<std::string>>&>())
+        .def("readWireReader",
+             [](VectorsCollectionMetadata& impl, yarp::os::idl::WireReader& reader) -> bool {
+                 return impl.read(reader);
+             })
+        .def("readConnectionReader",
+             [](VectorsCollectionMetadata& impl, yarp::os::ConnectionReader& connection) -> bool {
+                 return impl.read(connection);
+             })
+        .def("writeWireWriter",
+             [](const VectorsCollectionMetadata& impl, const yarp::os::idl::WireWriter& writer) -> bool {
+                 return impl.write(writer);
+             })
+        .def("writeConnectionWriter",
+             [](const VectorsCollectionMetadata& impl, yarp::os::ConnectionWriter& connection) -> bool {
+                 return impl.write(connection);
+             })
```

Comment:
This is not clear to me, in theory, there are no bindings of `yarp::os::ConnectionWriter` and `yarp::os::idl::WireReader` so you cannot call this method from python. How did you test them? moreover you can check this for function overloading in pybind11 https://pybind11.readthedocs.io/en/stable/classes.html#overloaded-methods

Reply:
Better to discuss face-to-face